presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntergrationMixedCase.java

```diff
@@ -0,0 +1,216 @@
+
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        this.server = new CassandraServer();
+        this.session = server.getSession();
+        createKeyspace(session, KEYSPACE);
+        return CassandraQueryRunner.createCassandraQueryRunner(server, ImmutableMap.of("case-sensitive-name-matching", "true"));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        server.close();
+    }
+
+    @Test
+    public void testCreateTable()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("cassandra")
+                .setSchema(KEYSPACE)
+                .build();
+        try {
+            getQueryRunner().execute(session, "CREATE TABLE TEST_CREATE(name VARCHAR(50), rollNum int)");
+            assertTrue(getQueryRunner().tableExists(session, "TEST_CREATE"));
+
+            getQueryRunner().execute(session, "CREATE TABLE  test_create(name VARCHAR(50), rollNum int)");
+            assertTrue(getQueryRunner().tableExists(session, "test_create"));
+
+            assertQueryFails(session, "CREATE TABLE TEST_CREATE (name VARCHAR(50), rollNum int)", "line 1:1: Table 'cassandra.test_connetor.TEST_CREATE' already exists");
+            assertFalse(getQueryRunner().tableExists(session, "Test"));
+        }
+        finally {
+            assertUpdate(session, "DROP TABLE IF EXISTS TEST_CREATE");
+            assertFalse(getQueryRunner().tableExists(session, "TEST_CREATE"));
+
+            assertUpdate(session, "DROP TABLE IF EXISTS test_create");
+            assertFalse(getQueryRunner().tableExists(session, "test_create"));
+        }
+    }
+
+    @Test
+    public void testCreateTableAs()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("cassandra")
+                .setSchema(KEYSPACE)
+                .build();
+        try {
+            getQueryRunner().execute(session, "CREATE TABLE TEST_CREATEAS AS SELECT * FROM tpch.region");
+            assertTrue(getQueryRunner().tableExists(session, "TEST_CREATEAS"));
+
+            getQueryRunner().execute(session, "CREATE TABLE IF NOT EXISTS test_createas AS SELECT * FROM tpch.region");
+            assertTrue(getQueryRunner().tableExists(session, "test_createas"));
+
+            getQueryRunner().execute(session, "CREATE TABLE TEST_CREATEAS_Join AS SELECT c.custkey, o.orderkey FROM " +
+                        "tpch.customer c INNER JOIN tpch.orders o ON c.custkey = o.custkey WHERE c.mktsegment = 'BUILDING'");
+            assertTrue(getQueryRunner().tableExists(session, "TEST_CREATEAS_Join"));
+
+            assertQueryFails("CREATE TABLE test_connector.TEST_CREATEAS_FAIL_Join AS SELECT c.custkey, o.orderkey FROM " +
+                        "tpch.customer c INNER JOIN tpch.ORDERS1 o ON c.custkey = o.custkey WHERE c.mktsegment = 'BUILDING'", "Table cassandra.tpch.ORDERS1 does not exist"); //failure scenario since tpch.ORDERS1 doesn't exist
```

Comment:
What is the significance of this test? Shouldn't we test with just `ORDERS` rather than `ORDERS1` as lowercase `orders` exists?